src/wavesurfer.js

```diff
@@ -1624,16 +1624,22 @@ export default class WaveSurfer extends util.Observer {
             peaks,
             val => Math.round(val * accuracy) / accuracy
         );
-        return new Promise((resolve, reject) => {
-            const json = JSON.stringify(arr);
 
+        return new Promise((resolve, reject) => {
+            const json = arr;
+            resolve(json);
+        }).then(response => {
             if (!noWindow) {
-                window.open(
-                    'data:application/json;charset=utf-8,' +
-                        encodeURIComponent(json)
+                let blobJSON = URL.createObjectURL(
+                    new Blob([JSON.stringify(response)], {
+                        type: 'application/json;charset=utf-8'
+                    })
                 );
+                window.open(blobJSON);
+                return blobJSON;
+            } else {
+                return response;
```

Comment:
We could do an early return here, something like this:
```
return new Promise((resolve, reject) => {
  if (noWindow) {
    return arr;
  }
  const blobJSON = URL.createObjectURL(...);
  window.open(blobJSON);
  return blobJSON;  
});
  ```

Reply:
yes we can do this as well,